### PR TITLE
Renamed and generalized package.

### PR DESCRIPTION
**Pull request by jimabramson**
@edx/ecommerce @mattdrayer FYI.  When this merges we'll bump the version, and will need to update the dependency references wherever in use.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,9 +5,9 @@
     long_description = readme.read()
 
 setup(
-    name='edx-ecommerce-api-client',
+    name='edx-rest-api-client',
     version='1.1.2',
-    description='Slumber client used to access APIs exposed by the edX E-Commerce Service',
+    description='Slumber client used to access various edX Platform REST APIs.',
```
Comment (mattdrayer) [inline setup.py]: Is it important to call-out that Slumber is being utilized in the description of this package?

File: setup.py [hunk 2]
```diff
@@ -5,9 +5,9 @@
     long_description = readme.read()
 
 setup(
-    name='edx-ecommerce-api-client',
+    name='edx-rest-api-client',
     version='1.1.2',
```
Comment (clintonb) [inline setup.py]: The version number needs to change so that there is no conflict with the old name+version combo.

**Comment by clintonb** [comment]
:+1: (once version number updated)

File: setup.py (hunk 1 above)
Comment (rlucioni) [inline setup.py]: @mattdrayer given that use of this API client is dictated entirely by [Slumber's conventions](http://slumber.readthedocs.org/en/v0.6.0/tutorial.html#using), I think calling this out is a good idea.

Comment (mattdrayer) [inline setup.py]: Cool, works for me, then

**Comment by mattdrayer** [comment]
:-1: +

**Comment by rlucioni** [comment]
@jimabramson once the version number is updated and this merges, you can release a new version to PyPI by pushing a new tag corresponding to the new version number. :+1:

**Comment by mattdrayer** [comment]
Would be cool if we could have Travis CI automatically push tags to PyPI...

**Comment by rlucioni** [comment]
@mattdrayer that's exactly what we've configured Travis to do here. All you need to do is push a tag to GitHub.

**Comment by mattdrayer** [comment]
MAGIC!!!  That is pretty awesome :)

**Comment by jimabramson** [comment]
FYI bumped version to 1.2.0

**Pull request merged**